examples/amm/src/contract.rs

```diff
@@ -31,33 +30,33 @@ impl Contract for Amm {
 
     async fn initialize(
         &mut self,
-        _runtime: &mut ContractRuntime,
+        runtime: &mut ContractRuntime<Abi>,
         _argument: (),
     ) -> Result<ExecutionOutcome<Self::Message>, AmmError> {
         // Validate that the application parameters were configured correctly.
-        assert!(Self::parameters().is_ok());
+        let _ = runtime.application_parameters();
 
         Ok(ExecutionOutcome::default())
     }
 
     async fn execute_operation(
         &mut self,
-        runtime: &mut ContractRuntime,
+        runtime: &mut ContractRuntime<Abi>,
         operation: Self::Operation,
     ) -> Result<ExecutionOutcome<Self::Message>, AmmError> {
         let mut outcome = ExecutionOutcome::default();
         if runtime.chain_id() == runtime.application_id().creation.chain_id {
-            self.execute_order_local(operation)?;
+            self.execute_order_local(runtime, operation)?;
         } else {
-            self.execute_order_remote(&mut outcome, operation)?;
+            self.execute_order_remote(runtime, &mut outcome, operation)?;
         }
 
         Ok(outcome)
     }
 
     async fn execute_message(
         &mut self,
-        runtime: &mut ContractRuntime,
+        runtime: &mut ContractRuntime<Abi>,
```

Comment:
It is not totally great to have a `runtime` being put all over. Could the runtime be part of the self? Not in this PR of course.

Reply:
Done, thanks!